indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -267,6 +271,18 @@ public SeekableStreamIndexTaskRunner(
     this.ingestionState = IngestionState.NOT_STARTED;
     this.lockGranularityToUse = lockGranularityToUse;
 
+    minMessageTime = ioConfig.getMinimumMessageTime().or(DateTimes.MIN);
+    maxMessageTime = ioConfig.getMaximumMessageTime().or(DateTimes.MAX);
+
+    if (ioConfig.getRefreshRejectionPeriodsInMinutes() != null) {
+      Execs.scheduledSingleThreaded("RejectionPeriodUpdater-Exec--%d")
```

Comment:
Who closes this exec service ?

Reply:
This seems to be creating on thread for each supervisor. Seems a bit much no ? Can we use a common exec service for this ?

Reply:
cc @adithyachakilam

Reply:
https://github.com/apache/druid/pull/17490

Reply:
> This seems to be creating on thread for each supervisor. Seems a bit much no ? Can we use a common exec service for this ?

+1

Reply:
this is creating thread on task not on supervisor ?